pkg/daemon/file_writers.go

```diff
@@ -46,11 +47,39 @@ func noOrigFileStampName(fpath string) string {
 }
 
 func createOrigFile(fromPath, fpath string) error {
-	if _, err := os.Stat(noOrigFileStampName(fpath)); err == nil {
-		// we already created the no orig file for this default file
-		return nil
+	orig := false
+
+	// https://issues.redhat.com/browse/OCPBUGS-11437
+	// MCO keeps the pull secret to .orig file once it replaced
+	// Adapt a check used in function deleteStaleData for backwards compatibility: basically if the file doesn't
+	// exist in /usr/etc (on FCOS/RHCOS) and no rpm is claiming it, we assume the orig file doesn't need to be created
+	if _, err := exec.Command("rpm", "-qf", fpath).CombinedOutput(); err == nil {
```

Comment:
suggestion (non-blocking): Taking a look at this RPM query and its analogue in the `deleteStaleData` function, the code flow seems very similar in both places. Given that, it would be nice improvement to consolidate the RPM query logic into a single function that both `createOrigFile()` and `deleteStaleData()` can use.